view/perspective/utils.js

```diff
@@ -386,25 +386,35 @@ function getValuesObject(accumulatorObject) {
       getPromiseWithUrl('/v1/subjects?fields=isPublished,absolutePath,tags'),
       getPromiseWithUrl('/v1/aspects?fields=isPublished,name,tags')
      ];
-     if (returnedPerspective) {
+
+     if (named) {
+      if (returnedPerspective) {
         setupSocketIOClient(returnedPerspective);
         valuesObj.perspective = returnedPerspective;
         valuesObj.name = valuesObj.perspective.name;
 
         // perspective exists. GET its hierarchy and lenses soon.
         promisesArr.concat(getPageLoadingPromises(returnedPerspective));
-     } else if (named) {
+      } else {
 
-      // named perspective does not exist
-      const name = url.split('/').pop();
-      customHandleError('Sorry, but the perspective you were trying ' +
-        'to load, ' + name + ', does not exist. Please select a ' +
-        'perspective from the dropdown.');
+        // named perspective does not exist
+        const name = url.split('/').pop();
+        customHandleError('Sorry, but the perspective you were trying ' +
+          'to load, ' + name + ', does not exist. Please select a ' +
+          'perspective from the dropdown.');
+      }
     }
 
     return Promise.all(promisesArr);
   })
   .then((responses) => {
+
+    // on redirect, responses is a string
+    // otherwise it is an array of API responses
```

Comment:
the problem was the returned redirect result, a string, came into the  last .then block as the variable named 'responses'. So the filter array failed on the string.